File: sdks/pulsar-bot-py/pulsar_bot/keyboard.py
"""Inline keyboard builder."""
from __future__ import annotations
from typing import List, Dict, Any


class Keyboard:
    """Helper for building inline-button keyboards."""

    def __init__(self) -> None:
        self._rows: List[List[Dict[str, str]]] = []

    @staticmethod
    def inline(buttons: List[List[Dict[str, str]]]) -> List[List[Dict[str, str]]]:
        """Build keyboard from nested list: [[{'text':..,'data':..}]]."""
        return [
            [{"text": b["text"], "callbackData": b["data"]} for b in row]
            for row in buttons
        ]

    def row(self, *buttons: Dict[str, str]) -> "Keyboard":
        self._rows.append(
            [{"text": b["text"], "callbackData": b["data"]} for b in buttons]
        )
        return self

    def build(self) -> List[List[Dict[str, str]]]:
        return self._rows

File: packages/bot-sdk/src/keyboard.ts
import type { InlineButton, InlineKeyboard } from './types.js';

export class Keyboard {
  private rows: InlineButton[][] = [];

  static inline(buttons: Array<Array<{ text: string; data: string }>>): InlineKeyboard {
    return buttons.map(row => row.map(b => ({ text: b.text, callbackData: b.data })));
  }

  row(...buttons: Array<{ text: string; data: string }>): this {
    this.rows.push(buttons.map(b => ({ text: b.text, callbackData: b.data })));
    return this;
  }

  build(): InlineKeyboard {
    return this.rows;
  }
}
